lib/reservation.rb

```diff
@@ -0,0 +1,23 @@
+module Hotel
+
+  class Reservation
+
+    attr_reader :date_range, :room_num, :cost
+
+    def initialize(date_range, room_num, rate: 200)
+      @date_range = date_range
+      @room_num = room_num
+      @rate = rate # constant for all normal reservations
+      @cost = get_cost
+
+      if !@room_num.to_s.match /\d+/ || @room_num > 20 || @room_num < 1
```

Comment:
Weirdly, my linter in Atom is saying that best Ruby style is to have the parens `(...)` around your `match` call (around the regex pattern), so it wants the line to look like this:
```ruby
if !@room_num.to_s.match(/\d+/) || @room_num > 20 || @room_num < 1
```